components/src/core/drawer/drawer-header.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { ReactNode, useCallback } from 'react';
+import { StyleSheet, View, Image } from 'react-native';
+import { H6, Subtitle } from '../typography';
+import {Divider, Theme, useTheme} from 'react-native-paper';
+
+
+const makeStyles = (props: DrawerHeaderProps, theme: Theme): any => StyleSheet.create({
+    root: {
+        backgroundColor: props.backgroundColor || theme.colors.primary
+    },
+    icon: {
+        display: 'flex',
+        height: 56,
+        width: 52,
+        paddingLeft: 4,
+        justifyContent: 'center',
+    },
+    content: {
+        color: 'red',
+        display: 'flex',
+        flexDirection: 'column',
+        padding: 4,
+        paddingLeft: 16,
+        flex: 1,
+        height: '100%',
+    },
+    title: {
+        color: props.fontColor || theme.colors.surface,
+        lineHeight: 30
+    },
+    subtitle: {
+        color: props.fontColor || theme.colors.surface,
+        lineHeight: 15,
+        marginTop: -2
+    },
+    backgroundImage: {
+        position: 'absolute',
+        width: '100%',
+        resizeMode: 'cover',
+        height: '100%',
+        opacity: props.backgroundOpacity,
+    },
+});
+
+export type DrawerHeaderProps = {
+    backgroundColor?: string;
+    backgroundImage?: string;
+    backgroundOpacity?: number;
+    fontColor?: string;
+    icon?: ReactNode;
+    title?: string;
+    subtitle?: string;
+    titleContent?: ReactNode;
```

Comment:
If we want to follow the react-native-paper pattern, we may want to have each our our components expose an optional `theme` prop for customizing the theme on individual components:
```
theme?: Theme;
```

then when we use `useTheme`, we would pass in the theme prop so that it gets merged with the theme from the Provider:
```
const theme = useTheme(props.theme);
```

For now we can probably ignore this, but we may want to log an enhancement issue for it to revisit.

Reply:
I think this ties in with the custom style overrides story that we've discussed.  I'm going to leave this out for now just so the scope is a bit more manageable.